quickwit/quickwit-search/src/lib.rs

```diff
@@ -340,3 +343,126 @@ pub fn searcher_pool_for_test(
             }),
     )
 }
+
+pub(crate) fn merge_resource_stats_it<'a>(
+    stats_it: impl IntoIterator<Item = &'a Option<ResourceStats>>,
```

Comment:
I think this was my code originally, but I suspect 

```suggestion
    stats_it: impl IntoIterator<Item = &'a ResourceStats>,
```
would be ok too.

Reply:
I changed this for `Option` because otherwise the caller side is in charge of handling the `None` case. This way, the policy for handling `None` is defined once and for all and tested.